cnd/src/network/comit_ln/swaps.rs

```diff
@@ -0,0 +1,602 @@
+        let mut swaps = Swaps::<()>::default();
+
+        let creation = swaps.create_as_pending_confirmation(
+            digest,
+            local_swap_id.clone(),
+            create_params.clone(),
+        );
+
+        assert!(creation.is_ok());
+
+        let created_swap = swaps.get_created_swap(&local_swap_id);
+
+        assert!(created_swap.is_some());
+
+        assert_eq!(created_swap.unwrap(), create_params)
+    }
+
+    #[test]
+    fn created_swap_as_pending_announcement_can_be_retrieved() {
+        let create_params = create_params();
+        let digest = create_params.clone().digest();
+        let local_swap_id = LocalSwapId::default();
+        let mut swaps = Swaps::<()>::default();
+
+        let creation = swaps.create_as_pending_announcement(
+            digest,
+            local_swap_id.clone(),
+            create_params.clone(),
+        );
+
+        assert!(creation.is_ok());
+
+        let created_swap = swaps.get_created_swap(&local_swap_id);
+
+        assert!(created_swap.is_some());
+
+        assert_eq!(created_swap.unwrap(), create_params)
+    }
+
+    #[test]
+    fn same_swap_cannot_be_created_twice_for_alice() {
+        let create_params = create_params();
+        let digest = create_params.clone().digest();
+        let local_swap_id = LocalSwapId::default();
+        let mut swaps = Swaps::<()>::default();
+
+        let _ = swaps.create_as_pending_confirmation(
+            digest.clone(),
+            local_swap_id.clone(),
+            create_params.clone(),
+        );
+        let creation = swaps.create_as_pending_confirmation(digest, local_swap_id, create_params);
+
+        assert!(creation.is_err())
+    }
+
+    #[test]
+    fn same_swap_cannot_be_created_twice_for_bob() {
+        let create_params = create_params();
+        let digest = create_params.clone().digest();
+        let local_swap_id = LocalSwapId::default();
+        let mut swaps = Swaps::<()>::default();
+
+        let _ = swaps.create_as_pending_announcement(
+            digest.clone(),
+            local_swap_id.clone(),
+            create_params.clone(),
+        );
+        let creation = swaps.create_as_pending_announcement(digest, local_swap_id, create_params);
+
+        assert!(creation.is_err())
+    }
+
+    #[test]
+    fn from_creation_to_finalisation_for_alice() {
+        let create_params = create_params();
+        let digest = create_params.clone().digest();
+        let local_swap_id = LocalSwapId::default();
+        let mut swaps = Swaps::<()>::default();
+
+        swaps
+            .create_as_pending_confirmation(digest.clone(), local_swap_id, create_params.clone())
+            .unwrap();
+
+        let shared_swap_id = SharedSwapId::default();
+        let (_local_swap_id, _create_params) = swaps
+            .move_pending_confirmation_to_communicate(&digest, shared_swap_id)
+            .unwrap();
+
+        assert_eq!(local_swap_id, _local_swap_id);
```

Comment:
Are these underscored because they are ignored? Should we give this a different name?